mapie/calibrators/ccp/utils.py

```diff
@@ -0,0 +1,534 @@
+    return functions
+
+
+def compile_functions_warnings_errors(
+    functions: List[Callable]
+) -> None:
+    """
+    Raise warnings and errors if the elements in ``functions`` have
+    unexpected arguments.
+
+    Raises
+    ------
+    ValueError
+        If functions contain unknown required arguments.
+
+    Warns
+    -----
+    UserWarning
+        If functions contain unknown optional arguments.
+
+    Notes
+    -----
+    This method ensures that the provided functions only use recognized
+    arguments ('X', 'y_pred', 'z'). Unknown optional arguments are allowed,
+    but will always use their default values.
+    """
+
+    warn_ind: Dict[str, List[int]] = {}
+    error_ind: Dict[str, List[int]] = {}
+    for i, funct in enumerate(functions):
+        assert callable(funct)
+        params = inspect.signature(funct).parameters
+
+        for param, arg in params.items():
+            if (
+                param not in ["X", "y_pred", "z"]
+                and param != "disable_marginal_guarantee"
+            ):
+                if arg.default is inspect.Parameter.empty:
+                    if param in error_ind:
+                        error_ind[param].append(i)
+                    else:
+                        error_ind[param] = [i]
+                else:
+                    if param in warn_ind:
+                        warn_ind[param].append(i)
+                    else:
+                        warn_ind[param] = [i]
+
+    if len(warn_ind) > 0:
+        warn_msg = ""
+        for param, inds in warn_ind.items():
+            warn_msg += (
+                f"The functions at index ({', '.join(map(str, inds))}) "
+                + "of the 'functions' argument, has an unknown optional "
+                + f"argument '{param}'.\n"
+            )
+        warnings.warn(
+            "WARNING: Unknown optional arguments.\n"
+            + warn_msg +
+            "The only recognized arguments are : 'X', 'y_pred' and 'z'. "
+            "The other optional arguments will act as parameters, "
+            "as it is always their default value which will be used."
+        )
+    if len(error_ind) > 0:
+        error_msg = ""
+        for param, inds in error_ind.items():
+            error_msg += (
+                f"The functions at index ({', '.join(map(str, inds))}) "
+                + "of the 'functions' argument, has an unknown required "
+                + f"argument '{param}'.\n"
+            )
+        raise ValueError(
+            "Forbidden required argument.\n"
+            f"{error_msg}"
+            "The only allowed required argument are : 'X', "
+            "'y_pred' and 'z'.\n"
+            "Note: You can use optional arguments if you want "
+            "to. They will act as parameters, as it is always "
+            "their default value which will be used."
+        )
+
+
+def sample_points(
+    X: ArrayLike,
+    points: Optional[Union[int, ArrayLike, Tuple[ArrayLike, ArrayLike]]]
```

Comment:
Was there a specific choice for calling this `points` and not `sample` (or something else)?

Reply:
I didn't really think about it... Should I rename it into ``sample`` ? I am open for new suggestions (@LacombeLouis, @thibaultcordier)

Reply:
It's a good question. I don't have strong convictions about it. @vincentblot28 @thibaultcordier any other ideas?

Reply:
to make it more explicit, why don't we call it "sample_centers" ?